lib/utility/recipe_puppy_parser.rb

```diff
@@ -0,0 +1,95 @@
+module Utility
+  class RecipePuppyParser
+    ALL_RECIPES_URL = 'http://www.recipepuppy.com/api/?q=&p='.freeze
+    BASE_URL = 'http://www.recipepuppy.com/api/'.freeze
+
+    def query_all_recipes_for(*ingredients)
+      ingredient_query_syntax = ingredients.join('').gsub(/\s+/, "")
+      contact_recipe_api(ingredient_query_syntax)
+    end
+
+    def contact_recipe_api(ingredient_query)
+      recipes = []
+      100.times do |page|
+        searched_url = "#{BASE_URL}?i=#{ingredient_query}&p=#{page + 1}"
+        get_recipes_to_query(recipes, searched_url)
+      end
+      recipes
+    end
+
+    def get_recipes_to_query(recipe_array, url)
+      response = HTTParty.get(url)
+      return error unless response.success?
+      parsed = JSON.parse(response)
+      parsed["results"].each do |recipe|
+        organise_new_vs_old_recipes(recipe)
+      end
+      recipe_array
+    end
+
+    def organise_new_vs_old_recipes(results_recipe)
+      ingredients = results_recipe["ingredients"].split(', ')
+      # Double check REGEX is working
+      recipe_name = results_recipe["title"]
+      recipe_name.gsub!(/&quot/, '').gsub!(/[^0-9A-Za-z]/, ' ')
+      check_recipes = Recipe.find_by(name: recipe_name)
+        process_ingredients_for(check_recipes, ingredients) if check_recipes
+
+      new_recipe = Recipe.create({
+        name: recipe_name,
+        thumbnail: results_recipe["thumbnail"]
+        })
+        process_ingredients_for(new_recipe, ingredients) if new_recipe
+    end
+
+    def process_ingredients_for(recipe, ingredients)
+      ingredients.each do |ingredient|
```

Comment:
If you would prefer not to use the conditional assignment format, please at least make your if condition a bit easier to read